### PR TITLE
Add new job state

### PR DESCRIPTION
**Pull request by coolo**
This is what I'm currently testing - just to mark my progress :)

### CODE REVIEW

**Comment by okurz** [comment]
looks good so far

**Comment by AdamWill** [comment]
one thing I think is probably worth checking with this approach: see what happens when we (try to) cancel jobs that are in this long-term blocked state.

**Comment by coolo** [comment]
while the worker is blocked uploading, it won't react to commands - so cancelling won't help. So possibly the webui shouldn't offer this action for this job state.

**Comment by coolo** [comment]
I noticed that we can't upload one single file for longer than 300 seconds because the API authentification is only updated for 300 seconds, so 1000 seconds is really catastrophy mode.

File: lib/OpenQA/WebSockets/Server.pm [hunk 1]
```diff
@@ -202,36 +204,52 @@ sub _message {
...
+    # running jobs have the default timeout
+    if ($job->state eq OpenQA::Schema::Result::Jobs::RUNNING) {
+        return 1;
+    }
+    # much bigger timeout for uploading jobs
+    if ($job->state eq OpenQA::Schema::Result::Jobs::UPLOADING) {
+        my $delta = DateTime->now()->epoch() - $job->worker->t_updated->epoch();
+        return if $delta > 1000;
+    }
+    return 0;
```
Comment (aaannz) [inline lib/OpenQA/WebSockets/Server.pm]: What about the last of execution states: waiting?

**Comment by AdamWill** [comment]
@coolo interactively through the web UI isn't the only way jobs get cancelled; what if I decide to re-run all the tests for a given ISO (by POST-ing it again) and one of them is in the middle of asset upload?

**Comment by aaannz** [comment]
In any case, worker will process cancel, and any other commands, only after upload is done. Even in "non blocking" upload, mojo doesn't process anything during file transfer. It only blocks later in ioloop.
My guess is that after upload it returns back to _stop_job without processing anything new in ioloop, that will finish all things related to job finish and undef $job, thus when cancel command is processed it is ignored because of not matching job.

File: lib/OpenQA/WebSockets/Server.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/WebSockets/Server.pm]: no timeout at all

**Comment by coolo** [comment]
the worker will continue uploading, no matter what state its job has.

But what worries me more: I was still not able to reproduce the 'Connection Error'. Our production instance produced it once today with a rather small file, while it used to produce it like mad for large images before the boundary hack was put back.

**Comment by coolo** [comment]
interesting. when I re-revert the revert I get the problem back - on first strike :)

**Comment by coolo** [comment]
http://mojolicious.org/perldoc/Mojo/IOLoop/Stream#timeout1 means we only have 15 seconds

**Comment by coolo** [comment]
But it's cancelled earlier: 

```
12:18:34.686685 write(1, "uploading test.qcow2\n", 21) = 21
12:18:34.687594 stat("/var/lib/openqa/pool/1/assets_public/test.qcow2", {st_mode=S_IFREG|0644, st_size=1029636096, ...}) = 0
12:18:34.687638 open("/var/lib/openqa/pool/1/assets_public/test.qcow2", O_RDONLY) = 7
12:18:36.020313 write(4, "ERROR test.qcow2: Connection err"..., 63) = 63
```

**Comment by coolo** [comment]
ah, but the last connect is from
```
 12:18:19.998452 connect(4, {sa_family=AF_INET, sin_port=htons(443), sin_addr=inet_addr("10.160.4.167")}, 16) = 0

**Comment by coolo** [comment]
https://github.com/os-autoinst/openQA/pull/848/commits/11fd87852c81f80d22b34b695666bd5214a8aab0 fixes this locally

File: lib/OpenQA/WebSockets/Server.pm (hunk 1 above)
Comment (coolo) [inline lib/OpenQA/WebSockets/Server.pm]: was this just curiousity or do you think I should mention it explicitly?

Comment (aaannz) [inline lib/OpenQA/WebSockets/Server.pm]: I wasn't sure if missing it was an intention. Maybe a comment that the rest is without a timeout?

Comment (coolo) [inline lib/OpenQA/WebSockets/Server.pm]: ah, right. I was thinking along the line, that a job in interactive should not be killed at all

**Comment by coolo** [comment]
d6ba8ec also takes waiting jobs into account for dead workers

**Comment by okurz** [comment]
some more tests would be nice

**Comment by coolo** [comment]
a very generic wisdom :)

**Comment by AdamWill** [comment]
From the commit message:

"Make UPLOADING the exception. We used to ignore WAITING jobs for dead workers, but this sounds unfair - these workers can die too"

will this cause interactive jobs that are waiting while a test admin creates a needle to get incompleted? Or will the test still be sending keepalives in that case? This was the case I assumed the `WAITING` exception existed for...

**Comment by aaannz** [comment]
Worker is sending the updates, thus acking it activity, even in waiting mode. Status updates are send by OpenQA::Worker, "waiting" is handled by isotovideo processes.

**Comment by coolo** [comment]
Right. But in both of our instances this dead worker detection kills *tons* of jobs - it seems the 10 seconds are just too eager

**Comment by AdamWill** [comment]
Isn't it a bit dangerous that the code to send status updates is supposed to send them exactly every 10 seconds and the check also uses exactly 10 seconds as its cutoff? Gives no margin for a test taking slightly longer to send its update for whatever reason (it was momentarily busy at the time the timer should've kicked in). Also delay in the status update actually being read by the server and the database updated. If you just look at the HTTP logs on a server, you do sometimes see an apparent 11 seconds between status updates for a job...I would've expected a bit more of an overlap, like the jobs update every 10 seconds and the check looks for ones which haven't updated in 15 or 20...

**Comment by coolo** [comment]
Yeah, it's suprising this didn't trigger a problem more often before - but then again I put the timer for the job from 20 minutes to 2 minutes, so we do *see* these problems.

**Pull request merged**